### PR TITLE
Update convolutional-neural-networks.md

### LINKED ISSUE #2133: [Term Entry] AI Covolution
### Reason/inspiration (optional)

We would like a new term entry for the concept of Convolution.

The entry should go in a new file in docs/content/ai/concepts/neural-networks/terms/convolution/convolution.md

The entry should
- Describe what Convolution is.
- Explain how it is used in a Convolutional Neural Network.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by YanisaHS**
### Description

I updated the Convolutional Neural Net page to include more information on the convolution process and inner workings of the network.

Closes #2133 

### Type of Change

- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Review by SSwiniarski — APPROVED** [review]
@YanisaHS This looks good. I am passing it on to a second review.

File: content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md [hunk 1]
```diff
@@ -12,16 +12,28 @@ CatalogContent:
...
+
+In a CNN's convolutional filter context, _f(x)_ represents the input data, and _g(x)_ represents the convolutional filter used in the network. The input data is a small matrix typically from an image or a feature map produced by a previous layer in the network, and the convolutional filter is a small matrix of weights that was learned in the training process. The convolutional filter essentially acts as a window that scans the input data to extract relevant features.
+
+During convolution, the filter (_g(x)_) is slid over the input data (_f(x)_). At each position, the filter performs element-wise multiplication of the two matrices. The products are then summed to produce a single value which is the output of the convolution in that position. This process is repeated for all positions, which results in an “output feature map,” which is a two-dimensional representation of the activations produced by applying the convolutional filters to the input data.
```
Comment (HishamT) [inline content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md:27]: Is it possible to show an example of this using a 2X2 matrix?

File: content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md [hunk 2]
```diff
@@ -12,16 +12,28 @@ CatalogContent:
...
+
+## Challenges of CNNs
+
+Although CNNs are commonly employed in computer vision tasks, there are also some challenges, such as:
 
-The output (extracted features from the CNN layer) are then passed on to fully connected layers, which aid the CNN in classification and prediction tasks.
+- Computational cost: Training CNNs typically requires substantial computational resources, and training with large datasets of high-resolution images can be time-consuming. Training times can also be long, and specialized hardware (e.g., GPUs) is often used to accelerate computations.
+- Need for large datasets: CNNs need large amounts of labeled training data in order to generalize well and learn meaningful features. Insufficient training data can lead to overfitting, which means the model fails to generalize to new, unseen data.
```
Comment (HishamT) [inline content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md:34]:
```suggestion
Need for large datasets: CNNs need large amounts of labeled training data to generalize well and learn meaningful features. Insufficient training data can lead to overfitting, which means the model fails to generalize to new, unseen data.
```

Comment (HishamT) [inline content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md:34]: Can this be rewritten without the use of the term "generalize"?It is not very clear what it means to be "generalized" from this context.

**Review by HishamT — COMMENTED** [review]
Hey @YanisaHS, I have finished reviewing your entry and left you some comments. 😃

File: content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md (hunk 2 above)
Comment (YanisaHS) [reply content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md]: I opted to add more info on "generalization" instead of omitting the term in my most recent commit. But, if you think it'd be better to remove the term "generalize", it could be changed to

"CNNs need large amounts of labeled training data to perform well and learn meaningful features. Insufficient training data can lead to overfitting, which means the model fails to make reliable predictions for data it hasn't been exposed to in the training process."

File: content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md (hunk 1 above)
Comment (YanisaHS) [reply content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md:27]: Do you have a recommendation of how this example might look? 😅 The description I provided is VERY simplified but illustrates a multi-step process with complicated steps, and I'm having a hard time coming up with an image that can summarize this while maintaining the simplicity of the description. (Or at least one that I know how to make)

My first thought was to have the example show an image, then pull out a rectangle of it and have that be the matrix (f(x)), and then have another matrix (g(x)) to show the matrix multiplying, but I'm not sure how to actually make something like that? (I usually use draw.io, but I'm open to trying another free tool if you have a recommendation!)

 I searched for some other images on how CNNs work, but most of them depict either a large overview of how CNNs work (with additional info I haven't gone into in this article), or are separated into multiple images with more complicated descriptions.

**Comment by YanisaHS** [comment]
@HishamT Sorry for the delay on this one - I've addressed your comments above. Thanks for the review!

File: content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md (hunk 1 above)
Comment (HishamT) [reply content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md:27]: It doesn't have to be an image. The convolution operation can be shown using a 2X2 matrix over a 5X5 matrix for example. Something like this: https://medium.com/@ramitag18/performing-convolution-on-a-matrix-4682fd364591

Comment (HishamT) [reply content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md:27]: You could include pictures of convolving two matrices together. 😁 LMK if this works. If it is too complicated, we can do without it.

**Review by HishamT — COMMENTED** [review]
More review comments

File: content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md (hunk 1 above)
Comment (YanisaHS) [reply content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md:27]: @HishamT Ok! I think I'd rather do without it, if that still works for you? Not because I don't like the idea, just because it's hard to make 😅

**Comment by YanisaHS** [comment]
@HishamT Ready for you to look at it again!

File: content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md (hunk 1 above)
Comment (HishamT) [reply content/ai/concepts/neural-networks/terms/convolutional-neural-networks/convolutional-neural-networks.md:27]: That's fine. 😃

**Review by HishamT — APPROVED** [review]
